packages/vizbuilder/src/components/AreaChart.jsx

```diff
@@ -0,0 +1,134 @@
+import React from "react";
+import {NonIdealState} from "@blueprintjs/core";
+
+import createChartConfig, {charts, ALL_YEARS} from "../helpers/chartconfig";
+import ChartCard from "./ChartCard";
+
+import "./_AreaChart.css";
+
+class AreaChart extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeType: null,
+      year: null
+    };
+
+    this.actions = Object.keys(charts).reduce((box, type) => {
+      box[type] = this.selectChart.bind(this, type);
+      return box;
+    }, {});
+
+    this.resizeCall = undefined;
+    this.scrollCall = undefined;
+
+    this.scrollEnsure = this.scrollEnsure.bind(this);
+    this.selectYear = this.selectYear.bind(this);
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return (
+      this.props.dataset !== nextProps.dataset ||
+      this.state.activeType !== nextState.activeType ||
+      this.state.year !== nextState.year
+    );
+  }
+
+  dispatchScroll() {
+    // TODO: Discuss how could we implement IntersectionObserver
+    window.dispatchEvent(new CustomEvent("scroll"));
+  }
+
+  dispatchResize() {
+    window.dispatchEvent(new CustomEvent("resize"));
+  }
```

Comment:
what do these `"scroll"` and `"resize"` events do? I believe they are not necessary.

Reply:
Since the vizbuilder uses a lot of charts at the same time, and there's a transition to change the size of the extended chart, sometimes the other charts aren't rendered after the user comes back to the multi-chart view. Also, when there's only 2 charts in a view, there's no scrollbar to ensure rendering if that happens, so these 2 functions are there as safeguards.
They're used only in specific `div`s, and have debouncers, so the overhead is minimal.